src/app/api/v1/user/update-comm-option/route.ts

```diff
@@ -33,9 +33,13 @@ export async function POST(req: NextRequest) {
       if (!subscriber) {
         throw new Error("No subscriber found for current session.");
       }
+      // -1 = Set as null, do not send instant notifications. Newly added in MNTOR-1368
       // 0 = Send breach alerts to the corresponding affected emails.
       // 1 = Send all breach alerts to user's primary email address.
-      const allEmailsToPrimary = Number(communicationOption) === 1 ?? false;
+      const allEmailsToPrimary =
+        Number(communicationOption) === -1
+          ? null
+          : Number(communicationOption) === 1 ?? false;
```

Comment:
I think the `?? false` is a but suspect here. The `Number(value) === 1` should already cast out to a Boolean value, which should mean the nullish coalescing shouldn't be reached since any invalid value to the `Number()` constructor would return as `NaN` instead of `null|undefined`.

```js
const options = [null, "-1", "0", "1", 2, "socks"];

for (const opt of options) {
  const toPrimary = Number(opt) === -1 ? null : Number(opt) === 1 ?? "nullish thing";
  console.log({ opt, cast: Number(opt), toPrimary });
}

// OUTPUT:
{opt: null, cast: 0, toPrimary: false}
{opt: '-1', cast: -1, toPrimary: null}
{opt: '0', cast: 0, toPrimary: false}
{opt: '1', cast: 1, toPrimary: true}
{opt: 2, cast: 2, toPrimary: false}
{opt: 'socks', cast: NaN, toPrimary: false}
```

Reply:
Not convinced this is any better than the double ternery. Meh.
```js
const options = [null, "-1", "0", "1", 2, "socks"];

for (const opt of options) {
  const toPrimary = opt === "-1" ? null : !!Number(opt);
  console.log({ opt, cast: Number(opt), toPrimary });
}

// OUTPUT:
{ opt: null, cast: 0, toPrimary: false }
{ opt: '-1', cast: -1, toPrimary: null }
{ opt: '0', cast: 0, toPrimary: false }
{ opt: '1', cast: 1, toPrimary: true }
{ opt: 2, cast: 2, toPrimary: true } // SLIGHTLY DIFFERENT BEHAVIOR DUE TO `!!` bool casting
{ opt: 'socks', cast: NaN, toPrimary: false }
```

Reply:
changed it to ternery